app/[locale]/roadmap/_components/ReleaseCarousel.tsx

```diff
@@ -0,0 +1,272 @@
+"use client"
+
+import { useEffect, useState } from "react"
+
+import { Image } from "@/components/Image"
+import { ButtonLink } from "@/components/ui/buttons/Button"
+import {
+  Carousel,
+  type CarouselApi,
+  CarouselContent,
+  CarouselItem,
+  CarouselNext,
+  CarouselPrevious,
+} from "@/components/ui/carousel"
+
+import { cn } from "@/lib/utils/cn"
+
+import { releasesData } from "@/data/roadmap/releases"
+
+const formatReleaseDate = (date: string) => {
+  if (/^\d{4}$/.test(date)) {
+    return date
+  }
+  return new Date(date).toLocaleDateString("en-US", {
+    month: "long",
+    day: "numeric",
+    year: "numeric",
+  })
+}
+
+const findLatestReleaseIndex = () => {
+  const today = new Date()
+  const twoMonthsFromNow = new Date()
+  twoMonthsFromNow.setMonth(today.getMonth() + 2)
+
+  // First try to find a release within the next 2 months
+  const upcomingReleaseIndex = releasesData.findIndex((release) => {
+    const releaseDate = new Date(release.releaseDate)
+    return releaseDate > today && releaseDate <= twoMonthsFromNow
+  })
+
+  // If no upcoming release found, find the most recent release up to today
+  if (upcomingReleaseIndex === -1) {
+    const pastReleases = releasesData.filter(
+      (release) => new Date(release.releaseDate) <= today
+    )
+    if (pastReleases.length > 0) {
+      const mostRecentRelease = pastReleases[pastReleases.length - 1]
+      return releasesData.findIndex(
+        (release) => release.releaseDate === mostRecentRelease.releaseDate
+      )
+    }
+  }
+
+  return upcomingReleaseIndex
+}
+
+const ReleaseCarousel = () => {
+  const todayDate = new Date()
+  const twoMonthsFromNow = new Date()
+  twoMonthsFromNow.setMonth(todayDate.getMonth() + 2)
+
+  const [api1, setApi1] = useState<CarouselApi>()
+  const [api2, setApi2] = useState<CarouselApi>()
+  const [currentIndex, setCurrentIndex] = useState(findLatestReleaseIndex())
```

Comment:
if the intention here is to have an initial value, we should call it with a function ([initialized function](https://react.dev/reference/react/useState#parameters)). Otherwise, `findLatestReleaseIndex` would be called in every render.
```suggestion
  const [currentIndex, setCurrentIndex] = useState(() => findLatestReleaseIndex())
```